lib/aws/ses.rb

```diff
@@ -9,15 +9,26 @@ class Base
       def initialize(*); end
 
       def deliver(mail)
-        ses_client.send_raw_email(raw_message: { data: mail.to_s }).tap do |response|
-          mail.header[:ses_message_id] = response.message_id
-        end
+        response = send_raw_email(mail)
+        mail.header[:ses_message_id] = response.message_id
+        response
       end
 
       alias deliver! deliver
 
       private
 
+      def send_raw_email(mail)
+        if IdentityConfig.store.ses_configuration_set_name.present?
+          ses_client.send_raw_email(
+            raw_message: { data: mail.to_s },
+            configuration_set_name: IdentityConfig.store.ses_configuration_set_name,
```

Comment:
re: the branching, does the client have a bad result with a nil value?

Reply:
the configuration would be `''` rather than `nil` unfortunately

Reply:
maybe a `.to_s` on there that casts `nil.to_s == ''`? (but otherwise not a big deal)

```suggestion
            configuration_set_name: IdentityConfig.store.ses_configuration_set_name.to_s,
```

Reply:
I tested quick and it seems like it's fine, so I'll call the method directly